software/glasgow/applet/memory/floppy/__init__.py

```diff
@@ -336,35 +335,36 @@ def __init__(self, pins):
     def elaborate(self, platform):
         m = Module()
 
-        m.d.comb += [
-            self.pins.redwc_t.oe.eq(1),
-            self.pins.redwc_t.o.eq(~self.redwc),
-            self.pins.motea_t.oe.eq(1),
-            self.pins.motea_t.o.eq(~self.mote[0]),
-            self.pins.drvsb_t.oe.eq(1),
-            self.pins.drvsb_t.o.eq(~self.drvs[1]),
-            self.pins.drvsa_t.oe.eq(1),
-            self.pins.drvsa_t.o.eq(~self.drvs[0]),
-            self.pins.moteb_t.oe.eq(1),
-            self.pins.moteb_t.o.eq(~self.mote[1]),
-            self.pins.dir_t.oe.eq(1),
-            self.pins.dir_t.o.eq(~self.dir),
-            self.pins.step_t.oe.eq(1),
-            self.pins.step_t.o.eq(~self.step),
-            self.pins.wdata_t.oe.eq(1),
-            self.pins.wdata_t.o.eq(~self.wdata),
-            self.pins.wgate_t.oe.eq(1),
-            self.pins.wgate_t.o.eq(~self.wgate),
-            self.pins.side1_t.oe.eq(1),
-            self.pins.side1_t.o.eq(~self.side1),
-        ]
-        m.submodules += [
-            FFSynchronizer(~self.pins.index_t.i, self.index),
-            FFSynchronizer(~self.pins.trk00_t.i, self.trk00),
-            FFSynchronizer(~self.pins.wpt_t.i, self.wpt),
-            FFSynchronizer(~self.pins.rdata_t.i, self.rdata),
-            FFSynchronizer(~self.pins.dskchg_t.i, self.dskchg),
-        ]
+        m.submodules.redwc_buffer = redwc_buffer = io.Buffer("o", self.ports.redwc)
+        m.d.comb += redwc_buffer.o.eq(~self.redwc)
```

Comment:
The new I/O system has a trick up its sleeve: you can tell it that the port itself is inverted (like how you can do it in the platform resource system, but without all the bulk of the latter).

```suggestion
        m.submodules.redwc_buffer = redwc_buffer = io.Buffer("o", ~self.ports.redwc)
        m.d.comb += redwc_buffer.o.eq(self.redwc)
```

If you do this then it's not necessary to invert all the accesses. Please do this where applicable.